test/cpp/jit/test_interpreter.cpp

```diff
@@ -1,9 +1,124 @@
 #include "test/cpp/jit/test_base.h"
 #include "test/cpp/jit/test_utils.h"
 
+#include <stdexcept>
 namespace torch {
 namespace jit {
 
+void testTypeCheck() {
+  {
+    auto graph = std::make_shared<Graph>();
+    std::unordered_map<std::string, Value*> vmap;
+    parseIR(
+        R"IR(
+graph(%a.1 : Tensor,
+      %b.1 : Tensor):
+  %t : Float(2:2, 2:1, device=cuda:0, requires_grad=1), %t0 : Float(3:3, 3:1), %type_matched : bool = prim::TypeCheck(%a.1, %b.1)
+  %14 : Tensor = prim::If(%type_matched)
```

Comment:
Why do we need this if in the test? Could we just
```
return (%t, %t0, %type_matched)
```
and check all the outputs, or at least
```
return (%type_matched)
```
if you would like to only check the boolean result? This `if` only obfuscates things for no reason IMO.

Reply:
I simplified the test. I think it is easier to read if a or b is returned. Hope you like it

Reply:
The test doesnt enforce semantics of the output values. We could be returning random tensors there and nothing will catch that in our tests. If we don't want to care what tensors we return, we should change the semantics of the op to not return any tensors at all - but if we return them we need to check what we return.
And honestly I don't really understand how 10 lines long test is easier to read than 4 lines test:
```
graph(%a : Tensor,
      %b : Tensor):
  %x : Float(2:2, 2:1, device=cuda:0, requires_grad=1), %y : Float(3:3, 3:1), %type_matched : bool = prim::TypeCheck(%a, %b)
  return (%x, %y, %type_matched)
```

Reply:
I agree. I will simplify the test.